### PR TITLE
api: fix bugs in score testing workflow

### PR DESCRIPTION
**Pull request by MJedr**
Ref. #1085
* cleanup lints

### CODE REVIEW

File: inspire_disambiguation/cli.py [hunk 1]
```diff
@@ -113,15 +108,24 @@ def ethnicity_model(load_data_path, save_model_path):
...
     click.secho("Starting Distance training.")
     test_signatures = train_and_save_distance_model(
-            ethnicity_model_path, save_model_path, sampled_pairs_size,
-            train_fraction)
+        ethnicity_model_path, save_model_path, sampled_pairs_size, train_fraction
+    )
     if test:
-        cluster(ethnicity_model_path, save_model_path, n_jobs, test_signatures)
+        click.secho("Starting clustering.")
+        cluster_from_redis(
```
Comment (monaawi) [inline inspire_disambiguation/cli.py]: I don't think this is the right way to go. ```cluster_from_redis``` is just a method to start clustering on signature blocks received from redis. For example, by calling it in the testing, you will end up consuming things from the redis queue.

**Review by monaawi — COMMENTED** [review]
Could you please write the exact error you were getting?

File: inspire_disambiguation/cli.py (hunk 1 above)
Comment (MJedr) [reply inspire_disambiguation/cli.py]: So before we used cluster method for that. We can go for that, it should work (without calculating statistics per each block, we'll get one global statistic describing the classification of the whole dataset)

File: inspire_disambiguation/api.py [hunk 2]
```diff
@@ -198,34 +218,43 @@ def cluster(
...
+def cluster_from_redis(
+    ethnicity_model_path, distance_model_path, n_jobs,
+):
     """
     Process all signature blocks from redis set (one by one).
     Args:
         ethnicity_model_path (str): Full path where ethnicity model is saved.
         distance_model_path (str): Full path where distance model is saved.
         n_jobs (int): How many jobs will be running to fit data.
+        test_signatures_uuids (list): Signature uuids which will be used
```
Comment (monaawi) [inline inspire_disambiguation/api.py]: this isn't needed anymore

File: inspire_disambiguation/api.py [hunk 3]
```diff
@@ -198,34 +218,43 @@ def cluster(
             ),
             signature_block=phonetic_block,
             B3_f_score=cluster.supervised_scoring(clusterer.y, cluster.labels_)
-            if hasattr(cluster, "supervised_scoring")
+            if hasattr(cluster, "supervised_scoring") and not test_labels
```
Comment (MJedr) [inline inspire_disambiguation/api.py:221]: I excluded only this statistic because others (training time, threshold) might be useful and I don't want to duplicate the code in the ```LOGGER.info()``` below

File: inspire_disambiguation/core/ml/models.py [hunk 4]
```diff
@@ -635,12 +624,14 @@ def load_data(self, signatures, input_clusters):
         self.X = np.empty((len(signatures_by_uuid), 1), dtype=np.object)
         self.y = -np.ones(len(self.X), dtype=np.int)
 
-        for i, cluster in enumerate(input_clusters):
+        i = 0
```
Comment (monaawi) [inline inspire_disambiguation/core/ml/models.py:627]: why did you change this?

Comment (MJedr) [reply inspire_disambiguation/core/ml/models.py:627]: I did the change to enumerate but it didn't work correctly with it - there is this continue statement in loop which should increase counter value and with enumerate it does it, what cause errors.

**Review by monaawi — APPROVED** [review]
